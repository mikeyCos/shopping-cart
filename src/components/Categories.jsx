import PropTypes from "prop-types";
import Loading from "./Loading";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
// import styles from "../styles/Categories.module.css";

const Categories = ({ categories }) => {
  console.log(categories);
  return (
    <article>
      {categories ? (
        <ul className="product-categories">
          {categories.map((category) => {
            return (
              <li key={category}>
                <Link to={`category/${category.replaceAll(" ", "-")}`}>
                  {category}
                </Link>
              </li>
            );
          })}
        </ul>
      ) : (
        <Loading />
      )}
    </article>
  );
};

Categories.propTypes = {
  categories: PropTypes.array,
};

export default Categories;
