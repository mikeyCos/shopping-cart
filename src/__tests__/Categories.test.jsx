import { describe, expect, it, vi } from "vitest";
import {
  render,
  screen,
  waitForElementToBeRemoved,
} from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Categories from "../components/Categories";
import { categories } from "./data.mocks";

describe("Categories component", () => {
  it("The Categories component with categories prop matches snapshot", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/shop"]}>
        <Categories categories={categories} />
      </MemoryRouter>
    );

    // const categoriesComponent = await screen.findByRole("article");
    // expect(categoriesComponent).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("The Categories component without categories prop renders Loading component", () => {
    render(
      <MemoryRouter initialEntries={["/shop"]}>
        <Categories />
      </MemoryRouter>
    );

    const categories = screen.getByTitle("Loading");
    expect(categories).toBeInTheDocument();
  });

  it("The categories list should have 5 items", () => {
    render(
      <MemoryRouter initialEntries={["/shop"]}>
        <Categories categories={categories} />
      </MemoryRouter>
    );

    const categoriesList = screen.getByRole("list");
    expect(categoriesList.children.length).toEqual(categories.length);
  });
});
