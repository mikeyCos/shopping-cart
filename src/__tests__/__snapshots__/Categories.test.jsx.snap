// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Categories component > The Categories component with categories prop matches snapshot 1`] = `
<div>
  <article>
    <ul
      class="product-categories"
    >
      <li>
        <a
          href="/category/all"
        >
          all
        </a>
      </li>
      <li>
        <a
          href="/category/electronics"
        >
          electronics
        </a>
      </li>
      <li>
        <a
          href="/category/jewelery"
        >
          jewelery
        </a>
      </li>
      <li>
        <a
          href="/category/men's-clothing"
        >
          men's clothing
        </a>
      </li>
      <li>
        <a
          href="/category/women's-clothing"
        >
          women's clothing
        </a>
      </li>
    </ul>
  </article>
</div>
`;
